cli/src/semgrep/formatter/sarif.py

```diff
@@ -14,12 +16,109 @@
 from semgrep.formatter.base import BaseFormatter
 from semgrep.rule import Rule
 from semgrep.rule_match import RuleMatch
+from semgrep.util import get_lines
 from semgrep.verbose_logging import getLogger
 
 logger = getLogger(__name__)
 
 
 class SarifFormatter(BaseFormatter):
+    @staticmethod
+    def _generate_taint_snippet(location: Any) -> Any:
+        content = "".join(
+            get_lines(Path(location.path), location.start.line, location.end.line + 2)
+        ).rstrip("\n")
+        snippet_before = "".join(
+            get_lines(
+                Path(location.path), location.start.line - 2, location.end.line - 1
+            )
+        ).lstrip("\n")
+        start_column = len(snippet_before) + location.start.col
+        end_column = len(snippet_before) + location.end.col
+        return "".join([snippet_before, content]), start_column, end_column
```

Comment:
That's a bit strange, since `start_column` no more represents a column (`snippet_before` can contain new lines, right?)

Reply:
Removed this part, I will create a separate PR for snippet extension